code/game/atoms_movable.dm

```diff
@@ -451,7 +452,7 @@
 					sleep(1)
 	else
 		var/error = dist_y/2 - dist_x
-		while(!gc_destroyed && target &&((((y < target.y && dy == NORTH) || (y > target.y && dy == SOUTH)) && dist_travelled < range) || isspaceturf(loc)) && throwing && istype(loc, /turf))
+		while(!gc_destroyed && target &&((((y < target.y && dy == NORTH) || (y > target.y && dy == SOUTH)) && dist_travelled < range) || isspaceturf(loc)) && (throwing||flying) && istype(loc, /turf))
```

Comment:
Wot they already be in throwing from the throw at making the flying unneeded?

Reply:
throwing is a flag that can be disabled mid flight, because you can catch/hit something. a flying flag ensure none of that happens